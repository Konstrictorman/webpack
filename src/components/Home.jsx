import React, {useState} from "react";

export const Home = (props) => {
   const {init} = props;
	const [counter, setCounter] = useState(init);

	const add = () => {
		setCounter(counter + 1);
	};

	const remove = () => {
		setCounter(counter - 1);
	};

	const reset = () => {
		setCounter(0);
	};

	return (
		<>
			<h1>Webpack App !!</h1>
			<button onClick={add}>Add</button>
			<button onClick={remove}>Remove</button>
			<button onClick={reset}>Reset</button>
			{counter}
		</>
	);
};
