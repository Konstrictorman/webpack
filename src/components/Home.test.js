import React from 'react';
import { Home } from './Home';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('Pruebas <Home/>', () => {
   let wrapper = renderer.create(<Home init={50}/>);
   let tree = wrapper.toJSON();
   expect(tree).toMatchSnapshot();
   
   
   beforeEach(() => {
      wrapper = renderer.create(<Home value={50}/>)
   })

   test('should return <Home> correctamente', () => {
      

      expect(wrapper).toMatchSnapshot();
   })


})