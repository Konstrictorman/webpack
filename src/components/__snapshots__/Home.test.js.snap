// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
[
  <h1>
    Webpack App !!
  </h1>,
  <button
    onClick={[Function]}
  >
    Add
  </button>,
  <button
    onClick={[Function]}
  >
    Remove
  </button>,
  <button
    onClick={[Function]}
  >
    Reset
  </button>,
  "50",
]
`;

exports[`Pruebas <Home/> should return <Home> correctamente 1`] = `
[
  <h1>
    Webpack App !!
  </h1>,
  <button
    onClick={[Function]}
  >
    Add
  </button>,
  <button
    onClick={[Function]}
  >
    Remove
  </button>,
  <button
    onClick={[Function]}
  >
    Reset
  </button>,
]
`;
